### ISSUE TITLE
NSInternalInconsistencyException after revoked app access

### ISSUE THREAD

**Issue by marcelmendesfilho** [microsoftgraph/msgraph-sdk-ios-nxoauth2-adapter#13]
My app is getting the following exception when executing the function connectToGraph().
The exception occurs in the following cenario:
0) I launch the app, authenticate on my Live account and authorize the app;
1) I revoke the app access to One Drive on my Live account;
2) After the access token expires, I launch the app, execute connectToGraph() and get the following error on console, but no errors are raised within the app or function. 

_2018-03-10 21:15:58.717407-0300 MyApp[8513:5555794] oauthConnection Error: {"error":"invalid_grant","error_description":"AADSTS70000: The request was denied because one or more scopes requested are unauthorized or expired. The user must first sign in and grant the client application access to the requested scope.
Trace ID: 80e43970-6a45-4124-8cb2-3f6490960900
Correlation ID: e86dee9b-88d3-444a-8967-437b01f225fd
Timestamp: 2018-03-11 00:15:58Z","error_codes":[70000],"timestamp":"2018-03-11 00:15:58Z","trace_id":"80e43970-6a45-4124-8cb2-3f6490960900","correlation_id":"e86dee9b-88d3-444a-8967-437b01f225fd"}_

3) Then I retry and execute connectToGraph() again. As the app does not get any errors, the next step is to set authentication provider.:
_MSGraphClient.setAuthenticationProvider(authentication.authenticationProvider)_

4) After that, the app sends a request to One Drive:
_graphClient.me().drive().items("root").children().request()_

It's when the exception occurs:

_2018-03-10 21:17:12.347195-0300 MyApp[8513:5555794] *** Assertion failure in -[NXOAuth2AuthenticationProvider appendAuthenticationHeaders:completion:], /Users/marcelmendesfilho/Hollywood/MyApp/Pods/MSGraphSDK-NXOAuth2Adapter/MSGraphSDKNXOAuth2/Auth/NXOAuth2AuthenticationProvider.m:239
2018-03-10 21:17:12.351426-0300 MyApp[8513:5555794] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'OAuth2AuthenticationProvider user account is missing access token'
*** First throw call stack:
(0x18670f164 0x185958528 0x18670f038 0x1870a97f4 0x10574ee58 0x1055e2f14 0x10556c920 0x1049f9d04 0x1049f9a7c 0x1049f6ef0 0x1049f6f54 0x18fce24a8 0x18fce221c 0x1900158f4 0x18ff37aa8 0x18ff2ae5c 0x18fcbc464 0x1866b6cdc 0x1866b4694 0x1866b4c50 0x1865d4c58 0x188480f84 0x18fd2d5c4 0x1049d1264 0x1860f456c)
libc++abi.dylib: terminating with uncaught exception of type NSException_

This is the connectToGraph() function where the exception is happening:

    func connectToGraph(withClientId clientId: String,
                                     scopes: [String],
                                     completion:@escaping (_ error: MSGraphError?) -> Void) {
    
        // Set client ID
        NXOAuth2AuthenticationProvider.setClientId(clientId, scopes: scopes)
        
        // Try silent log in. This will attempt to sign in if there is a previous successful
        // sign in user information.
        if NXOAuth2AuthenticationProvider.sharedAuth().loginSilent() == true {
            completion(nil)
        }
        // Otherwise, present log in controller.
        else {
            NXOAuth2AuthenticationProvider.sharedAuth()
                .login(with: nil) { (error: Error?) in
                    
                    if let nserror = error {
                        completion(MSGraphError.nsErrorType(error: nserror as NSError))
                    }
                    else {
                        completion(nil)
                    }
            }
        }
    }

Thanks for helping.

Marcel

**Comment by marcelmendesfilho** [comment]
Well... after some digging, I think I've found the cause, and it's not on adapter, but on NXOAuth2Client  itself:

NXOAuth2Account.m is posting a NXOAuth2AccountDidFailToGetAccessTokenNotification but to no observer. My understanding is that NXOAuth2AccountStore.m should observe it, as it is observing correlated notifications.

As a temporary solution, I added an NXOAuth2AccountDidFailToGetAccessTokenNotification observer into NXOAuth2AccountStore.m which calls accountDidLoseAccessToken function as it does another existing observer: NXOAuth2AccountDidLoseAccessTokenNotification. 

So... let's wait for the token to expire then see what happens.

Marcel
